### PR TITLE
[4.0] Split the rules field to 2 layouts

### PR DESCRIPTION
**Pull request by anuragteapot**
### Summary of Changes
Splits the HTML part (JLayout) into two files. 
Reasons:
1: less code per file
2: more efficient caching (less allocate memory, etc)

### Testing Instructions
1: Apply patch
2: Go to Admin->Global Configuration
3: Check Permissions working as before

### Expected result
This is code improvement

### Actual result
Same as before

### Documentation Changes Required
Yes

### CODE REVIEW

**Comment by anuragteapot** [comment]
Removed ACL tooltips due to remove in base branch.

**Comment by wilsonge** [comment]
Thanks!

**Comment by anuragteapot** [comment]
@wilsonge Thanks  :)

**Comment by Bakual** [comment]
Why was that Jlayout placed into `layouts/joomla/form/field/rules/tabs.php` and not like most other field layouts into `layouts/joomla/form/field/rules.php`.
Imho it looks like it was originally planned to have multiple layouts but in the end it was just one.

**Comment by anuragteapot** [comment]
@Bakual yeah, you are right.
I have planned for two but end up one  :p

**Comment by Bakual** [comment]
Thanks. I'll move it then in my PR :smile:

**Comment by anuragteapot** [comment]
@Bakual Thanks

File: layouts/joomla/form/field/rules/tabs.php [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+ * @var   boolean  $required        Is this field required?
+ * @var   integer  $size            Size attribute of the input.
+ * @var   boolean  $spellcheck      Spellcheck state for the form field.
+ * @var   string   $validate        Validation rules to apply.
+ * @var   string   $value           Value attribute of the field.
+ * @var   array    $checkedOptions  Options that will be set as checked.
+ * @var   boolean  $hasValue        Has this field a value assigned?
+ * @var   array    $options         Options available for this field.
+ *
+ * Calendar Specific
```
Comment (dgrammatiko) [inline layouts/joomla/form/field/rules/tabs.php:59]: Please make a PR to correct the variables used here

File: libraries/src/Form/Field/RulesField.php [hunk 2]
```diff
@@ -33,6 +33,14 @@ class RulesField extends FormField
 	 */
 	protected $type = 'Rules';
 
+	/**
+ 	 * Name of the layout being used to render the field
+ 	 *
+ 	 * @var    string
+	 * @since  3.5
```
Comment (dgrammatiko) [inline libraries/src/Form/Field/RulesField.php:40]: since not 3.5 but `__DEPLOY_VERSION__`

File: libraries/src/Form/Field/RulesField.php [hunk 3]
```diff
@@ -224,220 +216,42 @@ protected function getInput()
...
+		return trim($this->getRenderer($this->layout)->render($this->getLayoutData()));
+	}
 
-		return implode("\n", $html);
+	/**
+	 * Method to get the data to be passed to the layout for rendering.
+	 *
+	 * @return  array
+	 *
+	 * @since   3.5
```
Comment (dgrammatiko) [inline libraries/src/Form/Field/RulesField.php:237]: since not 3.5 but `__DEPLOY_VERSION__`

File: libraries/src/Form/Field/RulesField.php [hunk 4]
```diff
@@ -224,220 +216,42 @@ protected function getInput()
...
+		$extraData = array(
+			'groups'  		=>	$this->groups,
+			'section'		=>	$this->section,
+			'actions'		=>	$this->actions,
+			'assetId'		=>	$this->assetId,
+			'newItem'		=>	$this->newItem,
+			'assetRules'		=>	$this->assetRules,
+			'isGlobalConfig'	=>	$this->isGlobalConfig,
+			'parentAssetId'		=> 	$this->parentAssetId,
+		);
```
Comment (dgrammatiko) [inline libraries/src/Form/Field/RulesField.php:252]: All these variables need to be explained in the doc block of the JLayout

**Comment by dgrammatiko** [comment]
@Anu1601CS Please check my comments here and please make a PR to correct the doc blocks. Thanks

**Comment by anuragteapot** [comment]
@dgt41 Thanks for review PR coming soon.

File: layouts/joomla/form/field/rules/tabs.php (hunk 1 above)
Comment (anuragteapot) [reply layouts/joomla/form/field/rules/tabs.php:59]: should I remove JLayout variables which were not used in this file?

**Comment by florahwilliams** [comment]
Accounting assignment writing services have become very popular for students studying finance & accounting coursework writing services as they engage the best online [Accounting Writing Services](https://topessaywriting.com/accounting-writing-services-2/ "Accounting Writing Services"). <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19507">issues.joomla.org/tracker/joomla-cms/19507</a>.</sub>

**Pull request merged**